ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -1031,8 +1031,12 @@ public Path getPath() {
       return path;
     }
 
+    public boolean hasStatementId() {
```

Comment:
In case of statementId = -1, which means statement ID is not present, it returns 0
If statementId = 0, it returns 0.

So, isn't it coming to the original issue? Am I missing something?

Reply:
you mean that `getStatementId()` returns 0 in two cases:

- not set
- set to 0

?

yes. i think this could be changed too.

Reply:
yeah, we may need this information outside the Hive to figure out whether statementId is present for the delta.